src/middleware/authentication.ts

```diff
@@ -136,19 +137,30 @@ export class Authentication {
 			// Only for rules when it's not allowed for unauthorized users
 			// we need to find customer and assign it to the response.locals
 			if (!_resp.data.isAllowedUnauthorized) {
-				const user = await UserService.instance.get(_resp.data.userId);
-				if (!user) {
-					return response.status(StatusCodes.NOT_FOUND).json({
-						error: `Looks like user with logToId ${_resp.data.userId} is not found`,
-					});
+				if (_resp.data.userId !== '') {
```

Comment:
I think we could simplify this.
1. First of all, `response.locals.user` is used only for tracking purposes I think we can setup it only when `userId` is placed in `_resp`. Of course, there is a corner case for `/account/idtoken` endpoint, but we can add additional check. The corner case will be if the user tries to get the idToken using M2M token.
2. We can add `customerId` into `IAuthResponse` structure and return it in M2MUserInfoFetcher. Here the logic would be:
```
if (_resp.data.userId) {
    // fetch userEntity and setup to "response"
}
if (_resp.data.customerId) {
   // fetch customerEntity and fill the "response"
}
```
P.S. let's chat in case of having any questions

Reply:
My comment was not about comparing with empty string, but with changing the main logic. I just suggested to add `customerId` in the same manner as we use `userId` and setup what we got after fetching userInfo.
I can add such changes into your branch if you don't mind